### DISCUSSION TITLE
Feedback on a Python interpreter discovery mechanism

### DISCUSSION THREAD

**Discussion by brettcannon** [pypa/virtualenv#2345] [General]
I'm looking for any potential feedback on https://github.com/brettcannon/python-launcher/discussions/168 from virtualenv's perspective. Virtualenv could use this to help discover all the relevant Python installations on a machine. Virtualenv could also produce the information for environments it has created for tools that are looking for that information (e.g. Python Launcher, VS Code).

To be clear, this is in no way an ask to implement anything. I'm just asking if there is any visible holes in the JSON schema or idea from virtualenv's perspective. I would assume some library would get developed to help with this and potentially a PEP to help standardize it.

**Comment by gaborbernat** [comment]
Did you take a look at https://github.com/pypa/virtualenv/blob/main/src/virtualenv/discovery/py_info.py#L33-L112? All information available there would be relevant for virtualenv :thinking: As for search spec this file https://github.com/pypa/virtualenv/blob/main/src/virtualenv/discovery/py_spec.py#L21-L26 is relevant.

**Reply by brettcannon** [reply]
Thanks for the pointers!

**Comment by gaborbernat** [comment]
Thank you for seeking feedback from the virtualenv community on the Python interpreter discovery JSON schema. The referenced python-launcher discussion (https://github.com/brettcannon/python-launcher/discussions/168) is still open and has gathered extensive feedback from the ecosystem, and virtualenv's relevant code was pointed out in the response. Since this is primarily tracking an external discussion and not a virtualenv-specific question, I'm closing this discussion. If there are specific changes virtualenv should make based on the finalized schema, please open a new issue or discussion. Feel free to reach out if you need further input from the virtualenv maintainers.
